File: day3/Exercise_day3/supplementary_material_mpi/broadcast.f90
!----------------------------------------------------------------------   
!     program to demonstrate MPI_Bcast
!----------------------------------------------------------------------   

      use mpi 
 
      implicit none
 
      integer :: rank, data, ierror

!----------------------------------------------------------------------         
      call MPI_Init(ierror)
      call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierror)

      if (rank == 0) then
        write(*,*) 'enter a value:'
        read(*,*) data
      end if

!.....broadcast the value of data of rank 0 to all ranks

      write(*,*) "I am rank", rank, "and the value is", data

      call MPI_Finalize(ierror)
!----------------------------------------------------------------------   
      end program broadcast
!----------------------------------------------------------------------   

File: day3/Exercise_day3/supplementary_material_mpi/broadcast.cpp
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[])
{
    int rank, data;
    MPI_Init(&argc, &argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    if (rank==0) {
        printf("enter a value:\n");
        fflush(stdout);
        scanf ("%d",&data);
    }

    /* broadcast the value of data of rank 0 to all ranks */

    printf("I am rank %i and the value is %i\n", rank, data);
    MPI_Finalize();
    return 0;
}
